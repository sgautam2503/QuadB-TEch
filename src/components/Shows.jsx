import PropTypes from 'prop-types';
import {
  Button, Figure, Card, Col,
} from 'react-bootstrap';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { renderShowDetails } from '../redux/shows/showsSlice';

function Shows({
  id, name, rating, image,
}) {
  const dispatch = useDispatch();

  const handleShowDetails = () => {
    dispatch(renderShowDetails(id));
  };

  return (
    <>
      <Col xs={6} md={3} className="p-0">
        <Card
          className="odd"
          key={id}
          style={{
            color: '#fff',
            width: 'auto',
            borderColor: '#4369b2',
            borderRadius: '0',
          }}
        >
          <Card.Body className="d-flex flex-column align-items-center">
            <Figure>
              <Figure.Image alt={name} src={image} />
            </Figure>
            <Card.Title className="text-center">{name}</Card.Title>
            <Card.Subtitle className="mb-2">
              Rating:&nbsp;
              {rating}
            </Card.Subtitle>

            <Link to="/details">
              <Button className='details-btn'
                size="sm"
                style={{
                  width: '15vw',
                }}
                onClick={handleShowDetails}
              >
                <a>View Details</a>
              </Button>
            </Link>
          </Card.Body>
        </Card>
      </Col>
    </>
  );
}

Shows.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  rating: PropTypes.number,
  image: PropTypes.string,
}.isRequired;

export default Shows;
