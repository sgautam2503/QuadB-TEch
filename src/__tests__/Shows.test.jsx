import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import Shows from '../components/Shows';

describe('Shows component', () => {
  it('should display 2 shows on the screen', () => {
    const mockStore = configureStore([]);

    const initialState = {
      showsItems: [
        {
          id: 1,
          name: 'Under the Dome',
          type: 'Scripted',
          language: 'English',
          genres: ['Drama', 'Science-Fiction', 'Thriller'],
          status: 'Ended',
          ended: '2015-09-10',
          officialSite: 'http://www.cbs.com/shows/under-the-dome/',
          schedule: {
            time: '22:00',
            days: ['Thursday'],
          },
          rating: {
            average: 6.5,
          },
          network: {
            name: 'CBS',
            country: {
              name: 'United States',
            },
            officialSite: 'https://www.cbs.com/',
          },
          image: {
            medium:
              'https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg',
          },
          summary:
            "<p><b>Under the Dome</b> is the story of a small town that is suddenly and inexplicably sealed off from the rest of the world by an enormous transparent dome. The town's inhabitants must deal with surviving the post-apocalyptic conditions while searching for answers about the dome, where it came from and if and when it will go away.</p>",
        },
        {
          id: 2,
          name: 'Person of Interest',
          type: 'Scripted',
          language: 'English',
          genres: ['Action', 'Crime', 'Science-Fiction'],
          status: 'Ended',
          ended: '2016-06-21',
          officialSite: 'http://www.cbs.com/shows/person_of_interest/',
          schedule: {
            time: '22:00',
            days: ['Tuesday'],
          },
          rating: {
            average: 8.8,
          },
          network: {
            name: 'CBS',
            country: {
              name: 'United States',
            },
          },
          image: {
            medium:
              'https://static.tvmaze.com/uploads/images/medium_portrait/163/407679.jpg',
          },
          summary:
            "<p>You are being watched. The government has a secret system, a machine that spies on you every hour of every day. I know because I built it. I designed the Machine to detect acts of terror but it sees everything. Violent crimes involving ordinary people. People like you. Crimes the government considered \"irrelevant\". They wouldn't act so I decided I would. But I needed a partner. Someone with the skills to intervene. Hunted by the authorities, we work in secret. You'll never find us. But victim or perpetrator, if your number is up, we'll find you.</p>",
        },
      ],
    };

    const store = mockStore(initialState);
    render(
      <BrowserRouter>
        <Provider store={store}>
          {initialState.showsItems.map((show) => (
            <Shows
              key={show.id}
              id={show.id}
              name={show.name}
              rating={show.rating.average}
              image={show.image.medium}
              detailsStatus={show.detailsStatus}
            />
          ))}
        </Provider>
      </BrowserRouter>,
    );

    const shows = screen.getAllByRole('button');
    screen.debug();
    expect(shows).toHaveLength(2);
    expect(store.getState()).toMatchSnapshot();
  });
});
