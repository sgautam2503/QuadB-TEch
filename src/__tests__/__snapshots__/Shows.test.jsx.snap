// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shows component > should display 2 shows on the screen 1`] = `
{
  "showsItems": [
    {
      "ended": "2015-09-10",
      "genres": [
        "Drama",
        "Science-Fiction",
        "Thriller",
      ],
      "id": 1,
      "image": {
        "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
      },
      "language": "English",
      "name": "Under the Dome",
      "network": {
        "country": {
          "name": "United States",
        },
        "name": "CBS",
        "officialSite": "https://www.cbs.com/",
      },
      "officialSite": "http://www.cbs.com/shows/under-the-dome/",
      "rating": {
        "average": 6.5,
      },
      "schedule": {
        "days": [
          "Thursday",
        ],
        "time": "22:00",
      },
      "status": "Ended",
      "summary": "<p><b>Under the Dome</b> is the story of a small town that is suddenly and inexplicably sealed off from the rest of the world by an enormous transparent dome. The town's inhabitants must deal with surviving the post-apocalyptic conditions while searching for answers about the dome, where it came from and if and when it will go away.</p>",
      "type": "Scripted",
    },
    {
      "ended": "2016-06-21",
      "genres": [
        "Action",
        "Crime",
        "Science-Fiction",
      ],
      "id": 2,
      "image": {
        "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/163/407679.jpg",
      },
      "language": "English",
      "name": "Person of Interest",
      "network": {
        "country": {
          "name": "United States",
        },
        "name": "CBS",
      },
      "officialSite": "http://www.cbs.com/shows/person_of_interest/",
      "rating": {
        "average": 8.8,
      },
      "schedule": {
        "days": [
          "Tuesday",
        ],
        "time": "22:00",
      },
      "status": "Ended",
      "summary": "<p>You are being watched. The government has a secret system, a machine that spies on you every hour of every day. I know because I built it. I designed the Machine to detect acts of terror but it sees everything. Violent crimes involving ordinary people. People like you. Crimes the government considered \\"irrelevant\\". They wouldn't act so I decided I would. But I needed a partner. Someone with the skills to intervene. Hunted by the authorities, we work in secret. You'll never find us. But victim or perpetrator, if your number is up, we'll find you.</p>",
      "type": "Scripted",
    },
  ],
}
`;
